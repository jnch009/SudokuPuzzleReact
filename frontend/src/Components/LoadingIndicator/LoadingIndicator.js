import React from 'react';

const LoadingIndicator = () => {
  return (
    <div className='w-100 d-flex justify-content-center'>
      <div className='spinner-border' role='status'>
        <span className='sr-only'>Loading...</span>
      </div>
    </div>
  );
};

export default LoadingIndicator;
