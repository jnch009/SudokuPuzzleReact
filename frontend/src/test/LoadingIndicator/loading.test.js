import React from 'react';
import { render, screen } from '@testing-library/react';
import LoadingIndicator from '../../Components/LoadingIndicator/LoadingIndicator';

it('Loading Indicator visible', () => {
  const { container } = render(<LoadingIndicator />);
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild.firstChild.classList.contains('spinner-border')).toBeTruthy();
  expect(screen.queryByRole('status')).toBeTruthy();
});
