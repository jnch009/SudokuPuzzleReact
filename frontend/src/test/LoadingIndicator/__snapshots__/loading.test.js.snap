// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Indicator visible 1`] = `
<div
  class="w-100 d-flex justify-content-center"
>
  <div
    class="spinner-border"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
